### DISCUSSION TITLE
Datasource/terraform-provider: deprecate releases.hashicorp.com

### DISCUSSION THREAD

**Discussion by secustor** [renovatebot/renovate#25064] [Suggest an Idea]
### Tell us more.

The hunt strategy has been added in 2021 to cover provider which have not been published on the central Terraform registry. 

The missing provider then has been google-beta which made it in the meantime to the registry.    
https://registry.terraform.io/v1/providers/hashicorp/google-beta

This is in preparation for OpenTofu and Schema validation support to simplify it before. 
My proposal is to check how many request by the Renovate App happen to https://releases.hashicorp.com which are successful. This indicates that the provider has not been found at the registry, but is available on the releases page.   

Alternatively we can remove the releases repository code directly and monitor if user come back reporting issues.

**Comment by rarkins** [comment]
@nabeelsaabna can we count this in the app?

**Comment by nabeelsaabna** [comment]
Hi @secustor, @rarkins 

I'll check for counts
example `"http statistics"` message below

```json
{
	"name": "renovate",
	"level": 20,
	"logContext": "",
	"repository": "",
	"urls": {
		"https://registry.terraform.io/v1/providers/ekristen/awsextras (GET,200)": 1,
		"https://releases.hashicorp.com/terraform-provider-mysql/index.json (GET,406)": 1
	},
	"hostStats": {
		"api.github.com": {
			"requestCount": 32,
			"requestAvgMs": 300,
			"queueAvgMs": 0
		},
		"registry.terraform.io": {
			"requestCount": 1,
			"requestAvgMs": 52,
			"queueAvgMs": 1
		},
		"releases.hashicorp.com": {
			"requestCount": 1,
			"requestAvgMs": 348,
			"queueAvgMs": 0
		}
	},
	"totalRequests": 34,
	"time": "2023-10-22T12:19:52.420Z",
	"msg": "http statistics"
}
```

**Reply by secustor** [reply]
Do we get the http code of hosts too?

**Reply by nabeelsaabna** [reply]
in the URLS section, it will need some work
